common-healing.lic

```diff
@@ -20,239 +75,949 @@ $wound_map = {
+  # Given the line of text from the HEALTH command output
+  # that expresses items lodged into you,
+  # returns a map of any lodged items where the keys
+  # are the severity and the values are the list of wounds.
+  def parse_lodged_items(lodged_line)
+    lodged_items = Hash.new { |h, k| h[k] = [] }
+    if lodged_line
+      # Remove the tidbits at the start and end of the sentence that aren't pertinent.
+      lodged_line = lodged_line.gsub(/^You have\s+/, '').gsub(/\.$/, '')
+      lodged_line.split(',').map(&:strip).each do |wound|
+        # Do regex then look for the body part match.
+        wound =~ $DRCH_LODGED_BODY_PART_REGEX
+        body_part = Regexp.last_match.names.find { |x| Regexp.last_match[x.to_sym] }
+        body_part = Regexp.last_match[:part] if body_part == 'part'
+        # Check for the lodged severity.
+        severity = /\blodged\s+(.*)\s+into your\b/.match(wound)[1]
+        severity = $DRCH_LODGED_TO_SEVERITY_MAP[severity]
+        lodged_items[severity] << Wound.new(
+          body_part: body_part,
+          severity: severity
+        )
       end
-      wounds[part] += [{ 'part' => part, 'location' => Regexp.last_match(2), 'type' => Regexp.last_match(1), 'level' => $wound_map[Regexp.last_match(3)] }] if line =~ /(\w+) (\w+):.*\-\- (.*)\r/
     end
-    waitrt?
-    wounds
+    return lodged_items
   end
 
-  def bind_wound(part, person = 'my')
+  def bind_wound(body_part, person = 'my')
     snap = [DRC.left_hand, DRC.right_hand]
-    if /You \w+ remove/ =~ DRC.bput("tend #{person} #{part}", 'You work carefully at tending', 'That area has already been tended to', 'That area is not bleeding', 'You fumble', 'too injured for you to do that', 'You \w+ remove')
+    if /You \w+ remove/ =~ DRC.bput("tend #{person} #{body_part}", 'You work carefully at tending', 'That area has already been tended to', 'That area is not bleeding', 'You fumble', 'too injured for you to do that', 'You \w+ remove')
       DRCI.dispose(DRC.left_hand) if DRC.left_hand != snap.first
       DRCI.dispose(DRC.right_hand) if DRC.right_hand != snap.last
-      bind_wound(part, person)
+      bind_wound(body_part, person)
     end
     waitrt?
   end
 
-  def unwrap_wound(part)
-    DRC.bput("unwrap my #{part}", 'You unwrap your bandages') # Are there other messages?
-    pause 5
-    bind_wound(part)
+  def unwrap_wound(body_part, person = 'my')
+    DRC.bput("unwrap #{person} #{body_part}", 'You unwrap .* bandages')
+    waitrt?
+  end
+
+  # Skill check to tend a bleeding wound.
+  # If you're not skilled enough you may
+  # make things worse and incur long roundtime.
+  def skilled_to_tend_wound?(severity)
+    return false unless severity
+    if severity.is_a? String
+      severity = $DRCH_BLEED_RATE_TO_SEVERITY_MAP[severity]
+    end
+    # As of December 2020, these are guestimates.
+    # I was not aware of any definitive data.
+    # Please correct as appropriate, and be conservative.
+    severity_to_minskill_map = {
```

Comment:
What do you think about extracting this out of here to make the function more readable?

Reply:
Good idea, thanks